echo_client.py

```diff
@@ -0,0 +1,27 @@
+import socket
+import sys
+buffersize = 8
+
+
+def start_client(message=""):
+    client_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP
+        )
+    client_socket.connect(('127.0.0.1', 50000))
+    client_socket.sendall(message.encode('utf-8'))
```

Comment:
I know it seems odd, but try writing a test that attempts to send a byte string containing an encoded unicode string through this, like perhaps `'\xc3\xbcbersetzer'` (this is übersetzer encoded w/ utf-8).  

Because you are calling `encode` on the message without testing whether it needs encoding or not, you might end up calling it on a byte string.  If you do, Python will helpfully decode it to unicode first using `sys.defaultencoding`, which is ASCII.  This will not end well.  

Use caution when using Python 2.x `encode` or `decode`.  They are not safe.